src/main/java/hudson/plugins/sonar/action/SonarCacheAction.java

```diff
@@ -38,11 +38,11 @@
   private List<ProjectInformation> lastProjInfo;
 
   public SonarCacheAction() {
-    this.infoByTaskId = new HashMap<>();
+    this.infoByTaskId = new ConcurrentHashMap<>();
   }
 
   public List<ProjectInformation> get(SQProjectResolver resolver, long lastBuildTime, List<SonarAnalysisAction> analysis, Run<?, ?> run) {
-    if (lastRequest != null && age(lastRequest) < TimeUnit.SECONDS.toMillis(10)) {
+    if (lastRequest != null && age(lastRequest) < TimeUnit.SECONDS.toMillis(30)) {
```

Comment:
this gets called really frequently, just run a debugger with a breakpoint sitting on a project page...